common/src/main/java/whocraft/tardis_refined/common/tardis/TardisDesktops.java

```diff
@@ -4,71 +4,81 @@
+        return instance;
     }
 
-    @Override
-    protected void apply(Map<ResourceLocation, JsonElement> object, ResourceManager resourceManager, ProfilerFiller profilerFiller) {
-        DESKTOPS.clear();
-        object.forEach((resourceLocation, jsonElement) -> {
-            try {
-
-                JsonObject desktopObject = jsonElement.getAsJsonObject();
-
-                // Build the DesktopTheme
-                String id = desktopObject.get("id").getAsString();
-                String structure = desktopObject.get("structure").getAsString();
-                String name = desktopObject.get("name_component").getAsString();
-                ResourceLocation resourceLocationId = new ResourceLocation(id);
-
-                if (!resourceLocationId.equals(TardisDesktops.DEFAULT_OVERGROWN_THEME.getIdentifier())) {
-                    DesktopTheme theme = new DesktopTheme(resourceLocationId, new ResourceLocation(structure));
-                    theme.setName(name);
-                    addDesktop(theme);
-                }
+    public static CodecJsonReloadListener<DesktopTheme> getReloadListener(){
+        return RELOAD_LISTENER;
+    }
 
+    public static Map<ResourceLocation,DesktopTheme> getRegistry() {
+        return RELOAD_LISTENER.getData();
+    }
 
-            } catch (JsonParseException jsonParseException) {
-                TardisRefined.LOGGER.debug("Issue parsing Desktop {}! Error: {}", resourceLocation, jsonParseException.getMessage());
-            }
-        });
+    public static DesktopTheme getDesktopById(ResourceLocation location) {
+        return RELOAD_LISTENER.getData().getOrDefault(location, TardisDesktops.FACTORY_THEME);
+    }
 
-        if (Platform.getServer() != null) {
-            new SyncDesktopsMessage(DESKTOPS).sendToAll();
-        }
+    /**
+     * Creates and adds the Tardis Refined default list of DesktopThemes to a standalone map.
+     * Can be used for datagenerators or as a fallback registry
+     * @return
+     */
+    public static Map<ResourceLocation,DesktopTheme> registerDefaultDesktops() {
+        addDefaultDesktop(TardisDesktops.FACTORY_THEME);
+        addDefaultDesktop(new DesktopTheme("coral", "desktop/coral"));
+        addDefaultDesktop(new DesktopTheme("victorian","desktop/victorian"));
+        addDefaultDesktop(new DesktopTheme("copper", "desktop/copper"));
+        addDefaultDesktop(new DesktopTheme("toyota", "desktop/toyota"));
+        addDefaultDesktop(new DesktopTheme("crystal", "desktop/crystal"));
+        addDefaultDesktop(new DesktopTheme("nuka", "desktop/nuka"));
+        addDefaultDesktop(new DesktopTheme("future_nostalgia", "desktop/future_nostalgia"));
+        addDefaultDesktop(new DesktopTheme("violet_eye", "desktop/violet_eye"));
+        return DEFAULT_DESKTOPS;
+    }
 
+    private static void addDefaultDesktop(DesktopTheme theme) {
+//        TardisRefined.LOGGER.info("Adding default Desktop {}", theme.getIdentifier());
```

Comment:
Is this going to stay commented?

Reply:
Could be good to keep it in for when we need to debug. Normally we don't want this in players' logs because this can get called as a fallback dataset when the sync packet is initially sent, so excess log spam can occur.

Reply:
Is there methodology for only having this execute in the development environment?

Reply:
Platform.isProduction

Reply:
I'd recommend we have logs like that in that method

Reply:
Cool, I can modify it so that the logs show for development environments only.